libavcodec/qsvdec: skip non-key frame after "seek" function

wenbinc-Bin opened the pull request:
Fix #9095. Qsv decoder assume that after calling seek funcion, the first
frame should be key frame. However this is not true for some videos. If
the frame is not key frame after seek(), there will error. Conditional
statements are added to skip these frame until reading a key frame.

@xhaihao @xuguangxin @feiwan1 please help me to review.

thanks

xuguangxin commented on libavcodec/qsvdec.c:
```diff
@@ -616,6 +616,8 @@ int ff_qsv_process_data(AVCodecContext *avctx, QSVContext *q,
     }
 
     if (!q->initialized) {
+        if (ret < 0)
```
is it possible to move  this to line 616?

wenbinc-Bin replied:
I think no. If this is moved out of this conditional statement.  It will skip all the non-key frames all the time. It only need to skip frames when it reinitialize decoder.

xuguangxin replied:
better add some comment here, explain why we not return before this

wenbinc-Bin replied:
Ok

wenbinc-Bin committed:
libavcodec/qsvdec: skip non-key frame after "seek" function

Fix #9095. Qsv decoder assume that after calling seek funcion, the first
frame should be key frame. However this is not true for some videos. If
the frame is not key frame after seek(), there will error. Conditional
statements are added to skip these frame until reading a key frame.

Signed-off-by: Wenbin Chen <wenbin.chen@intel.com>

wenbinc-Bin commented:
submit to community.
